packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -0,0 +1,201 @@
+%lex
+%options case-insensitive
+
+%x string
+%x md_string
+%x title
+%x open_directive
+%x type_directive
+%x arg_directive
+%x close_directive
+%x acc_title
+%x acc_descr
+%x acc_descr_multiline
+%s axis_data
+%s data
+%s data_inner
+%%
+\%\%\{                                    { this.begin('open_directive'); return 'open_directive'; }
+<open_directive>((?:(?!\}\%\%)[^:.])*)    { this.begin('type_directive'); return 'type_directive'; }
+<type_directive>":"                       { this.popState(); this.begin('arg_directive'); return ':'; }
+<type_directive,arg_directive>\}\%\%      { this.popState(); this.popState(); return 'close_directive'; }
+<arg_directive>((?:(?!\}\%\%).|\n)*)      return 'arg_directive';
+\%\%(?!\{)[^\n]*                          /* skip comments */
+[^\}]\%\%[^\n]*                           /* skip comments */
+<axis_data>(\r?\n)                        { this.popState(); return 'NEWLINE'; }
+<data>(\r?\n)                             { this.popState(); return 'NEWLINE'; }
+[\n\r]+                                   return 'NEWLINE';
+\%\%[^\n]*                                /* do nothing */
+
+title                                     { this.begin("title"); return 'title'; }
+<title>(?!\n|;|#)*[^\n]*                  { this.popState(); return "title_value"; }
+
+
+accTitle\s*":"\s*                         { this.begin("acc_title");return 'acc_title'; }
+<acc_title>(?!\n|;|#)*[^\n]*              { this.popState(); return "acc_title_value"; }
+accDescr\s*":"\s*                         { this.begin("acc_descr");return 'acc_descr'; }
+<acc_descr>(?!\n|;|#)*[^\n]*              { this.popState(); return "acc_descr_value"; }
+accDescr\s*"{"\s*                         { this.begin("acc_descr_multiline");}
+<acc_descr_multiline>[\}]                 { this.popState(); }
+<acc_descr_multiline>[^\}]*               { return "acc_descr_multiline_value"; }
+
+"xychart-beta"		                        {return 'XYCHART';}
+("vertical"|"horizontal")                 {return 'CHART_ORIENTATION'}
+
+"x-axis"                                  { this.begin("axis_data"); return "X_AXIS"; }
+"y-axis"                                  { this.begin("axis_data"); return "Y_AXIS"; }
+<axis_data>[\[]                           { this.popState(); return 'SQUARE_BRACES_START'; }
+<axis_data>[+-]?\d+(?:\.\d+)?             { return 'NUMBER_WITH_DECIMAL'; }
+<axis_data>"-->"                          { return 'ARROW_DELIMITER'; }
+
+
+"line"                                    { this.begin("data"); return 'LINE'; }
+"bar"                                     { this.begin("data"); return 'BAR'; }
+<data>[\[]                                { this.popState(); this.begin("data_inner"); return 'SQUARE_BRACES_START'; }
+<data_inner>[+-]?\d+(?:\.\d+)?            { return 'NUMBER_WITH_DECIMAL';}
+<data_inner>[\]]                          { this.popState(); return 'SQUARE_BRACES_END'; }
```

Comment:
That is not clear why we would need other state than `data`
Replacing state `data` with `data_inner` is not necessary even if we need another one, because we have a stack of states, not only one:
```
<data>"[" { this.pushState('data_inner'); return 'SQUARE_BRACES_START'; }

Reply:
Removed